sycl/unittests/pi/EventTest.cpp

```diff
@@ -89,7 +89,105 @@ TEST_F(DISABLED_EventTest, PICreateEvent) {
             PI_SUCCESS);
 }
 
-TEST_F(DISABLED_EventTest, piEventGetInfo) {
+struct callback_user_data {
+  pi_int32 event_type;
+  int index;
+};
+
+TEST_F(EventTest, piEventSetCallback) {
+
+  constexpr size_t event_type_count = 3;
+
+  static bool triggered_flag[event_type_count] = {false, false, false};
+
+  pi_int32 event_callback_types[event_type_count] = {
+      PI_EVENT_SUBMITTED, PI_EVENT_RUNNING, PI_EVENT_COMPLETE};
+
+  callback_user_data user_data[event_type_count];
+
+  auto callback = [](pi_event event, pi_int32 status, void *data) {
+    ASSERT_NE(data, nullptr);
+
+    callback_user_data *pdata = static_cast<callback_user_data *>(data);
+
+#ifndef NDEBUG
+    printf("\tEvent callback %d of type %d triggered\n", pdata->index,
+           pdata->event_type);
+#endif
+
+    triggered_flag[pdata->index] = true;
+  };
+
+  // gate event lets us register callbacks before letting the enqueued work be
+  // executed.
+  pi_event gateEvent;
+  ASSERT_EQ((Plugins[0].call_nocheck<detail::PiApiKind::piEventCreate>(
+                _context, &gateEvent)),
+            PI_SUCCESS);
+
+  constexpr const size_t dataCount = 10000u;
```

Comment:
Please, reduce number of elements to speed up testing.

Reply:
I've reduced it to 1000.